File: apps/web/lib/types/bank.ts
export type LoanType = "home" | "investment" | "personal" | "car" | "business";

export interface BankBranding {
  primary_color: string;
  logo_url: string;
}

export interface LoanProduct {
  id: string;
  name: string;
  type: LoanType;
  interest_rate: string;
  min_amount: string;
  max_amount: string;
  min_term_months: number;
  max_term_months: number;
}

export interface Bank {
  id: string;
  name: string;
  slug: string;
  branding: BankBranding;
  products: LoanProduct[];
}

File: services/api/schemas/bank.py
import uuid
from decimal import Decimal

from pydantic import BaseModel, ConfigDict

from models.enums import LoanType


class BankBranding(BaseModel):
    primary_color: str
    logo_url: str


class LoanProductOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    name: str
    type: LoanType
    interest_rate: Decimal
    min_amount: Decimal
    max_amount: Decimal
    min_term_months: int
    max_term_months: int


class BankOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    name: str
    slug: str
    branding: BankBranding
    products: list[LoanProductOut]
